src/components/GraphiQL.js

```diff
@@ -694,6 +702,18 @@ export class GraphiQL extends React.Component {
     editor.setValue(print(parse(editor.getValue())));
   };
 
+  handleMergeQuery = () => {
+    const editor = this.getQueryEditor();
+    const query = editor.getValue();
+
+    if (!query) {
+      return;
```

Comment:
This is too forgiving - if merging fragments into a query isn't possible, we need to either show that it's not possible and/or disable this feature. A couple of scenarios where this feature isn't meaningful imo:
- No query string, like this code
- A query string without fragments - no fragments to merge then
- An invalid query string - this will break the query editor and ultimately GraphiQL as a whole

I think we **at least** need to handle the last "invalid query string" scenario. Could we do:
- catch the invalid query string and either just return (if we're getting rid of the button) and/or
- change the UI (I think "prettify" feature does something to make the button red)?